Compile failure on ubuntu 20.04 arm64

Hello, I am trying to compile S2H2 on Ubuntu 20.04 arm 64 using gcc-10 and/or clang-18, using (e.g.):

```
CC=clang-18 CXX=clang++-18 cmake .. -GNinja -DUSE_OPENGLES=1 \
 -DBUILD_CROWD_CONTROL=0 -DCMAKE_BUILD_TYPE=Release
cmake --build . -j8
```

With both compilers, the build goes a long way but fails in ActorBehavior.cpp line 136.

gcc:
```
In file included from /root/2ship2harkinian/mm/2s2h/Rando/StaticData/StaticData.h:6,
                 from /root/2ship2harkinian/mm/2s2h/Rando/Rando.h:4,
                 from /root/2ship2harkinian/mm/2s2h/Rando/ActorBehavior/ActorBehavior.h:4,
                 from /root/2ship2harkinian/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp:1:
/root/2ship2harkinian/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp: In function 'void Rando::ActorBehavior::OnFileLoad()':
/root/2ship2harkinian/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp:136:48: error: cannot convert 'void(GIVanillaBehavior, bool*, void*)' to 'GameInteractor::ShouldVanillaBehavior::fn' {aka 'std::function<void(GIVanillaBehavior, bool*, __va_list)>'}
  136 |     COND_HOOK(ShouldVanillaBehavior, IS_RANDO, MiscVanillaBehaviorHandler);
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                |
      |                                                void(GIVanillaBehavior, bool*, void*)
/root/2ship2harkinian/mm/2s2h/GameInteractor/GameInteractor.h:748:91: note: in definition of macro 'COND_HOOK'
  748 |             hookId = GameInteractor::Instance->RegisterGameHook<GameInteractor::hookType>(body); \
...
```

clang:
```
/root/2ship2harkinian/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp:136:5: error: no matching member function for call to 'RegisterGameHook'
  136 |     COND_HOOK(ShouldVanillaBehavior, IS_RANDO, MiscVanillaBehaviorHandler);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/2ship2harkinian/mm/2s2h/GameInteractor/GameInteractor.h:748:48: note: expanded from macro 'COND_HOOK'
  748 |             hookId = GameInteractor::Instance->RegisterGameHook<GameInteractor::hookType>(body); \
      |                      ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/2ship2harkinian/mm/2s2h/GameInteractor/GameInteractor.h:406:13: note: candidate function template not viable: no known conversion from 'void (GIVanillaBehavior, bool *, void *)' to 'typename GameInteractor::ShouldVanillaBehavior::fn' (aka 'function<void (GIVanillaBehavior, bool *, __builtin_va_list)>') for 1st argument
  406 |     HOOK_ID RegisterGameHook(typename H::fn h) {
      |             ^                ~~~~~~~~~~~~~~~~
```
I was able to compile the 1.1.2 release using clang, so this has arisen more recently

--- a/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp
+++ b/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp
@@ -7,7 +7,7 @@ extern "C" {
 }
 
 // This is kind of a catch-all for things that are simple enough to not need their own file.
-void MiscVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, void* optionalArg) {
+void MiscVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, va_list optionalArg) {
     switch (id) {
         case VB_MADAME_AROMA_ASK_FOR_HELP:
             *should = !CHECK_WEEKEVENTREG(WEEKEVENTREG_BOMBERS_NOTEBOOK_EVENT_RECEIVED_KAFEIS_MASK);

